client/lib/application/src/application-list/application-list.js

```diff
@@ -6,16 +6,31 @@ class AppDetail extends BaseView {
     constructor(props) {
         super({
             className: 'applicationList',
-            store: props.flux.getStore('application')
+            store: props.flux.getStore('application'),
+            events: {
+                'submit': 'filter'
+            }
         });
+        this.state = {
+            filterTerm: ''
+        };
+
     }
 
     update() {
         this.data = {
-            apps: this.store.getApplications()
+            apps: this.store.getApplications(this.state.filterTerm),
+            filterTerm: this.state.filterTerm
         };
     }
 
+    filter(evt) {
+        evt.preventDefault();
+        this.state.filterTerm = this.$el.find('input').val();
+        this.update();
+        this.render();
+    }
+
     render() {
         this.$el.html(Template(this.data));
         return this;
```

Comment:
You probably want fo re-focus the input here as well as reset the cursor properly.